test/DomainController.ts

```diff
@@ -0,0 +1,104 @@
+import { ethers } from "hardhat";
+import { Contract, ContractFactory } from "ethers";
+import { expect } from "chai";
+
```

Comment:
// tests
// for all tests
// mixed all together is it difficult  to understand what is the purpose of the test
// as best practice you can separate them by using describe
// also no test for event with expected value
// - 3 point